home/views/team_formation.py

```diff
@@ -0,0 +1,307 @@
+from django.contrib.admin.views.decorators import staff_member_required
+from django.contrib.auth.decorators import permission_required
+from django.core.paginator import Paginator
+from django.db.models import Prefetch
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import get_object_or_404, redirect, render
+from django.views.decorators.http import require_http_methods
+
+from home.dataclasses import ApplicantData, DjangonautDetail, TeamStatistics
+from home.filters import ApplicantFilterSet
+from home.forms import BulkTeamAssignmentForm, OverlapAnalysisForm
+from home.models import Session, SessionMembership, Team, UserSurveyResponse
+from home.utils.availability import (
+    calculate_team_overlap,
+    format_availability_by_day,
+)
+
+
+@permission_required("Team.form_team")
+@staff_member_required
+@require_http_methods(["GET", "POST"])
+def team_formation_view(request: HttpRequest, session_id: int) -> HttpResponse:
+    """
+    Main view for team formation interface.
+
+    Displays:
+    - Applicant list with filtering and pagination
+    - Quick team assignment actions
+    - Current teams with statistics
+    """
+    session = get_object_or_404(Session, pk=session_id)
+
+    # Handle bulk team assignment
+    if request.method == "POST":
+        bulk_form = BulkTeamAssignmentForm(request.POST, session=session)
+        if bulk_form.is_valid():
+            assigned_count = bulk_form.save()
+            team = bulk_form.cleaned_data["team"]
+            messages.success(
+                request,
+                f"Successfully assigned {assigned_count} user(s) to team '{team.name}'.",
+            )
+            return redirect("admin:session_form_teams", session_id=session.id)
+        # If form is invalid, errors will be displayed in the template
+    else:
+        bulk_form = BulkTeamAssignmentForm(session=session)
+
+    # Build applicant queryset with filtering
+    applicants_list, filterset = get_filtered_applicants(session, request.GET)
+
+    # Handle sorting
+    sort_by = request.GET.get("sort", "selection_rank")
+    sort_order = request.GET.get("order", "asc")
+
+    if sort_by in ["score", "selection_rank", "previous_application_count"]:
+        # Sort the list (it's a list of ApplicantData dataclass instances)
+        # Nulls always go last regardless of sort order
+        reverse = sort_order == "desc"
+
+        if reverse:
+            # Descending: highest values first, nulls last
+            applicants_list = sorted(
+                applicants_list,
+                key=lambda x: (
+                    getattr(x, sort_by) is None,
+                    -(getattr(x, sort_by) or 0),
+                ),
+            )
+        else:
+            # Ascending: lowest values first, nulls last
+            applicants_list = sorted(
+                applicants_list,
+                key=lambda x: (getattr(x, sort_by) is None, getattr(x, sort_by) or 0),
+            )
```

Comment:
Why is the sorting based on a list of python objects? This should ideally happen in the database.